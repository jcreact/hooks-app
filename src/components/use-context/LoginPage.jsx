import React, { useContext } from 'react';
import { useForm } from '../../hooks/useForm';
import { UserContext } from './UserContext';

export const LoginPage = () => {
    const [{ email, name, password }, handleInputChange, reset] = useForm({
        email: 'palmahn@gmail.com',
        name: 'Jos√© Palma',
        password: '',
    });

    const { setUser } = useContext(UserContext);

    /**
     * @param {React.FormEvent<HTMLFormElement>} evt
     */
    const handleSubmit = (evt) => {
        evt.preventDefault();
        setUser({ id: new Date().getTime(), name, email });
        reset();
        document.querySelector(`input[name:'email']`).select();
    };

    return (
        <div
            className="d-flex flex-column justify-content-center align-items-center"
            style={{ height: '90vh' }}
        >
            <div
                className="border border-1 border-dark bg-dark shadow"
                style={{ width: '40vw', borderRadius: '8px' }}
            >
                <form className="my-4 mx-3" onSubmit={handleSubmit}>
                    <div className="form-floating mb-3">
                        <input
                            type="text"
                            className="form-control"
                            placeholder="."
                            autoComplete="off"
                            autoFocus
                            name="email"
                            value={email}
                            onChange={handleInputChange}
                        />
                        <label>Email</label>
                    </div>
                    <div className="form-floating mb-4">
                        <input
                            type="password"
                            className="form-control"
                            placeholder="."
                            autoComplete="off"
                            name="password"
                            value={password}
                            onChange={handleInputChange}
                        />
                        <label>Password</label>
                    </div>
                    <div className="d-grid col-6 mx-auto">
                        <button
                            type="submit"
                            className="btn btn-outline-secondary btn-lg text-light"
                            style={{ borderRadius: '25px' }}
                        >
                            Login
                        </button>
                    </div>
                </form>
            </div>
        </div>
    );
};
