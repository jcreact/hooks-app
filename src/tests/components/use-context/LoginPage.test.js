import React from 'react';
import { mount } from 'enzyme';

//Componente
import { LoginPage } from '../../../components/use-context/LoginPage.jsx';

//Hook
import { UserContext } from '../../../components/use-context/UserContext.js';

describe(`Pruebas en el componente 'LoginPage.jsx'.`, () => {

    const setUser = jest.fn();

    const user = {
        id: expect.any(Number),
        email: 'palmahn@gmail.com',
        name: 'José Palma'
    };

    const component = mount(
        <UserContext.Provider value={{ setUser }} >
            <LoginPage />
        </UserContext.Provider>
    );

    test(`Debe renderizar el componente HomePage correctamente.`, () => {
        expect(component).toMatchSnapshot();
    });

    test(`Debe ejecutar la función 'setUser' con el usuario por defecto.`, () => {
        component.find('form').simulate('submit', { preventDefault() { } });
        expect(setUser).toBeCalled();
        expect(setUser).toBeCalledWith(user);
    });

});
