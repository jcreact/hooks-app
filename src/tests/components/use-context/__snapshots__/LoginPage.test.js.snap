// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente 'LoginPage.jsx'. Debe renderizar el componente HomePage correctamente. 1`] = `
<div
  className="d-flex flex-column justify-content-center align-items-center"
  style={
    Object {
      "height": "90vh",
    }
  }
>
  <div
    className="border border-1 border-dark bg-dark shadow"
    style={
      Object {
        "borderRadius": "8px",
        "width": "40vw",
      }
    }
  >
    <form
      className="my-4 mx-3"
      onSubmit={[Function]}
    >
      <div
        className="form-floating mb-3"
      >
        <input
          autoComplete="off"
          autoFocus={true}
          className="form-control"
          name="email"
          onChange={[Function]}
          placeholder="."
          type="text"
          value="palmahn@gmail.com"
        />
        <label>
          Email
        </label>
      </div>
      <div
        className="form-floating mb-4"
      >
        <input
          autoComplete="off"
          className="form-control"
          name="password"
          onChange={[Function]}
          placeholder="."
          type="password"
          value=""
        />
        <label>
          Password
        </label>
      </div>
      <div
        className="d-grid col-6 mx-auto"
      >
        <button
          className="btn btn-outline-secondary btn-lg text-light"
          style={
            Object {
              "borderRadius": "25px",
            }
          }
          type="submit"
        >
          Login
        </button>
      </div>
    </form>
  </div>
</div>
`;
